ml_metadata/tools/mlmd_bench/fill_context_edges_workload.cc

```diff
@@ -0,0 +1,245 @@
+
+// Adds a new context edge to `request`. Its properties will be set according to
+// `non_context_node_id` and `context_node_id`.
+template <typename T>
+void SetCurrentContextEdge(const int64 non_context_node_id,
+                           const int64 context_node_id,
+                           PutAttributionsAndAssociationsRequest& request) {
+  if (std::is_same<T, Attribution>::value) {
+    Attribution* context_edge = request.add_attributions();
+    context_edge->set_artifact_id(non_context_node_id);
+    context_edge->set_context_id(context_node_id);
+  } else if (std::is_same<T, Association>::value) {
+    Association* context_edge = request.add_associations();
+    context_edge->set_execution_id(non_context_node_id);
+    context_edge->set_context_id(context_node_id);
+  }
+}
+
+// Generates `num_edges` context edges within `request`.
+// Selects `num_edges` non-context and context node pairs from
+// `existing_non_context_nodes` and `existing_context_nodes` w.r.t. to the
+// categorical distribution with a Dirichlet prior. Also, performs rejection
+// sampling to ensure current generated pair is unique. Returns detailed error
+// if query executions failed.
+template <typename T>
+tensorflow::Status GenerateContextEdges(
+    const std::vector<Node>& existing_non_context_nodes,
+    const std::vector<Node>& existing_context_nodes, const int64 num_edges,
+    std::discrete_distribution<int64>& non_context_node_index_dist,
+    std::discrete_distribution<int64>& context_node_index_dist,
+    MetadataStore& store,
+    std::unordered_map<int64, std::unordered_set<int64>>& unique_checker,
+    PutAttributionsAndAssociationsRequest& request, int64& curr_bytes) {
+  CHECK((std::is_same<T, Attribution>::value ||
+         std::is_same<T, Association>::value))
+      << "Unexpected Types";
+  int64 i = 0;
+  std::default_random_engine gen;
+  while (i < num_edges) {
+    const int64 non_context_node_index = non_context_node_index_dist(gen);
+    const int64 context_node_index = context_node_index_dist(gen);
+    int64 non_context_node_id;
+    if (std::is_same<T, Attribution>::value) {
+      non_context_node_id =
+          absl::get<Artifact>(
+              existing_non_context_nodes[non_context_node_index])
+              .id();
+    } else if (std::is_same<T, Association>::value) {
+      non_context_node_id =
+          absl::get<Execution>(
+              existing_non_context_nodes[non_context_node_index])
+              .id();
+    }
+    const int64 context_node_id =
+        absl::get<Context>(existing_context_nodes[context_node_index]).id();
+
+    bool already_existed_in_current_setup =
+        CheckDuplicateContextEdgeInCurrentSetUp(
+            non_context_node_id, context_node_id, unique_checker);
+    tensorflow::Status status = CheckDuplicateContextEdgeInDb<T>(
+        non_context_node_id, context_node_id, store);
+    if (!status.ok() && status.code() != tensorflow::error::ALREADY_EXISTS) {
+      return status;
+    }
```

Comment:
drop case

Reply:
Yes, for this one, it maybe necessary since in the tests, I also included a case where the db has pre-inserted context edges. So, we should also perform a checking inside db to pass the tests. Wdyt?